### ISSUE TITLE
Javatar fails on installing Javatar package Java standard edition 8

### ISSUE THREAD

**Issue by Tankenstein** [spywhere/Javatar#43]
Whenever i start sublime, javatar does it's thing but then fails on installing Javatar package.  Unfortunately the error stays in the gutter for too little time, so i can't copy the whole thing, but it says ErrCode 2, and something about a file missing. The fix to this right now is every time i want to do java, i open sublime and reinstall javatar. It still throws that error, but it works for that session.

I'm on elementary OS 0.3 Freya 64 bit. (Ubuntu based)

**Comment by spywhere** [comment]
Could you try install it manually again by go to *Package Manager... > Install Package*? 

And if possible, please send me a History Actions (not a custom one) via Ctrl+Shift+K, Ctrl+Shift+H.

**Comment by Tankenstein** [comment]
Tried installing, same deal. History actions is this:
```
## Javatar Report
### System Informations
* Javatar Version: `1.0.0`
* Sublime Version: `3080`
* Package Path: `/home/uku/.config/sublime-text-3/Packages`
* Javatar Channel: `stable`
* Sublime Channel: `stable`
* Is Debug Mode: `False`
* Platform: `linux`
* As Packages: `True`
* Package Control: `True`
* Architecture: `x64`
* Javatar's Parent Folder: `Javatar`
* Is Project: `True`
* Is File: `True`
* Is Java: `True`

### Action List
1. Startup
2. Reset all settings
3. Reset all snippets
4. Reset all default packages
5. Read settings
6. Load snippets
7. Check news
8. Ready
9. Javatar snippet Class.javatar loaded
10. Analyse snippet [file=Packages/Javatar/snippets/Class.javatar]
11. Javatar snippet Enumerator.javatar loaded
12. Analyse snippet [file=Packages/Javatar/snippets/Enumerator.javatar]
13. Javatar snippet Interface.javatar loaded
14. Analyse snippet [file=Packages/Javatar/snippets/Interface.javatar]
15. Load Java default packages
16. Javatar default package Proto.javatar-packages loaded
17. Analyse package [file=Packages/Javatar/Developers/Proto.javatar-packages]
18. Check packages update
19. Install default package
20. Install Package [type=remote_package, name=Java Standard Edition 8]
21. Select item ['Javatar Settings...', 'Adjust Javatar settings'] [from_sublime=True]
22. Exit menu [from_sublime=False]
23. Select item ['Packages Manager...', 'Javatar packages manager'] [from_sublime=True]
24. Select item ['Install Packages...', 'Download and install new packages'] [from_sublime=False]
25. Select item ['Java Standard Edition 8', 'Ready to download (~4MB).'] [from_sublime=False]
26. Install Package [type=remote_package, name=Java Standard Edition 8]
27. Select item ['Actions History', 'Generate a report on Javatar actions history'] [from_sublime=False]

```

**Comment by spywhere** [comment]
Is there any error messages in Sublime Text's console? If yes, please send it too.

**Comment by Tankenstein** [comment]
No errors during normal operation or trying to install that package, but when i reinstall javatar with package control, this happens:
```
reloading plugin Javatar.Javatar
Traceback (most recent call last):
  File "/opt/sublime_text/sublime_plugin.py", line 116, in reload_plugin
    m.plugin_loaded()
  File "Javatar in /home/uku/.config/sublime-text-3/Installed Packages/Javatar.sublime-package", line 18, in plugin_loaded
  File "Javatar in /home/uku/.config/sublime-text-3/Installed Packages/Javatar.sublime-package", line 13, in startup
  File "utils.javatar_utils in /home/uku/.config/sublime-text-3/Installed Packages/Javatar.sublime-package", line 187, in hide_status
  File "utils.javatar_status in /home/uku/.config/sublime-text-3/Installed Packages/Javatar.sublime-package", line 41, in hide_status
  File "utils.javatar_validator in /home/uku/.config/sublime-text-3/Installed Packages/Javatar.sublime-package", line 19, in is_java
TypeError: argument of type 'NoneType' is not iterable
Traceback (most recent call last):
  File "/opt/sublime_text/sublime_plugin.py", line 127, in reload_plugin
    el.on_activated(v)
  File "utils.javatar_event_handler in /home/uku/.config/sublime-text-3/Installed Packages/Javatar.sublime-package", line 290, in on_activated
  File "utils.javatar_event_handler in /home/uku/.config/sublime-text-3/Installed Packages/Javatar.sublime-package", line 192, in on_activated
  File "Javatar in /home/uku/.config/sublime-text-3/Installed Packages/Javatar.sublime-package", line 26, in on_change
TypeError: 'NoneType' object is not callable
reloading Packages/Javatar/Javatar.sublime-settings
generating syntax summary
Javatar snippet Class.javatar loaded
reloading Packages/User/Package Control.sublime-settings
reloading Packages/User/Preferences.sublime-settings
Javatar snippet Enumerator.javatar loaded
Javatar snippet Interface.javatar loaded
[Javatar] Use default Java version [JDK1.8.0u40]
reloading Packages/User/Javatar.sublime-settings
reloading Packages/User/Preferences.sublime-settings
Traceback (most recent call last):
  File "/opt/sublime_text/sublime_plugin.py", line 304, in on_activated
    callback.on_activated(v)
  File "utils.javatar_event_handler in /home/uku/.config/sublime-text-3/Installed Packages/Javatar.sublime-package", line 290, in on_activated
  File "utils.javatar_event_handler in /home/uku/.config/sublime-text-3/Installed Packages/Javatar.sublime-package", line 192, in on_activated
  File "Javatar in /home/uku/.config/sublime-text-3/Installed Packages/Javatar.sublime-package", line 26, in on_change
TypeError: 'NoneType' object is not callable
```

**Comment by spywhere** [comment]
Could you paste your Javatar's user settings? Seem like there's a problem with key missing...

**Comment by Tankenstein** [comment]
```
{
	"javatar_gp": 0,
	"jdk_version":
	{
		"use": ""
	},
	"message_id": 20
}
```

There's also a project_data with my stuff, do i need to include that as well?

**Comment by spywhere** [comment]
If you can, yes (you can copy only something relevant to Javatar, not required to have it all).

**Comment by Tankenstein** [comment]
Here's the full thing.
```
{
	"javatar_gp": 0,
	"jdk_version":
	{
		"use": ""
	},
	"message_id": 20,
	"project_data":
	{
		"2":
		{
			"folders":
			[
				{
					"follow_symlinks": true,
					"path": "/home/uku/Soft/java/school/EX10"
				}
			],
			"javatar":
			{
				"dependencies":
				[
					"/home/uku/Soft/java/school/EX10/lib/BookStoreApiV01.jar",
					"/home/uku/Soft/java/school/EX10/lib/dom4j-1.6.1.jar"
				],
				"dependencies_path": "/home/uku/Soft/java/school/EX10/lib",
				"source_folder": "/home/uku/Soft/java/school/EX10/src"
			}
		}
	}
}
```

**Comment by spywhere** [comment]
Hmm. That's strange. Could you try fresh install Sublime Text, Package Control and Javatar and see if the error still persist? And if error still there, please try install it manually (download from the release, extract and paste in the proper packages path). If it still error, try install the JavaSE8 package from [here](https://github.com/spywhere/JavatarPackages/tree/master/packages) (copy and paste it anywhere within packages path).

If you can, although it's not necessary, try record and reproduce the error.

Right now, the error is not relevant to the problem you specified (packages installation failed) but it's a user's settings instead. So, I still have no idea what's going on.

Sorry for the inconvenient you might have.

**Comment by cmnth** [comment]
Same issue on Linux Mint and ST3. Fresh install - same Error 2. Reinstalling ST3/Javator or manually copying 'JavaSE8.javatar-packages' to '~/.config/sublime-text-3/Installed Packages' folder didn't resolve either.

**Comment by spywhere** [comment]
No, "JavaSE8.javatar-packages" should be installed inside "User" directory not "Installed Packages".

**Comment by m-delvalle** [comment]
In case it's still relevant:
I installed 'Javatar' recently, and had this same error after install. I didn't notice at first, but then when tried to "manually" (through Javatar's menu) install some packages, I got 'Failed to install Java SE8. Directory [...]/sublime-text-3/Packages/[...] doesn't exist'.
After a few tries and some frustration, I happened to see that the "error" path included '../user/'. Checking Sublime config folder, it's 'User' (with capital U). Created a new 'user' folder, and installation succeded.
So, there you go. There's s typo in your path configuration ;)

**Comment by caesarivs** [comment]
Same issue here, ST3 on Ubuntu 14.04.3, here's Javatar Actions History Report:

```
## Javatar Report
### System Informations
* Javatar Version: `1.0.1`
* Sublime Version: `3083`
* Package Path: `/home/caesarivs/.config/sublime-text-3/Packages`
* Javatar Channel: `stable`
* Sublime Channel: `stable`
* Is Debug Mode: `False`
* Platform: `linux`
* As Packages: `True`
* Package Control: `True`
* Architecture: `x64`
* Javatar's Parent Folder: `Javatar`
* Is Project: `True`
* Is File: `False`
* Is Java: `False`

### Action List
1. Startup
2. Reset all settings
3. Reset all snippets
4. Reset all default packages
5. Read settings
6. Load snippets
7. Check news
8. Ready
9. Javatar snippet Class.javatar loaded
10. Analyse snippet [file=Packages/Javatar/snippets/Class.javatar]
11. Javatar snippet Enumerator.javatar loaded
12. Analyse snippet [file=Packages/Javatar/snippets/Enumerator.javatar]
13. Javatar snippet Interface.javatar loaded
14. Analyse snippet [file=Packages/Javatar/snippets/Interface.javatar]
15. Load Java default packages
16. Javatar default package Proto.javatar-packages loaded
17. Analyse package [file=Packages/Javatar/Developers/Proto.javatar-packages]
18. Check packages update
19. Install default package
20. Install Package [type=remote_package, name=Java Standard Edition 8]
21. Select item ['Actions History', 'Generate a report on Javatar actions history'] [from_sublime=True]
```

Sublime's console:

```
DPI scale: 1
startup, version: 3083 linux x64 channel: stable
executable: /opt/sublime_text/sublime_text
working dir: /
packages path: /home/caesarivs/.config/sublime-text-3/Packages
state path: /home/caesarivs/.config/sublime-text-3/Local
zip path: /opt/sublime_text/Packages
zip path: /home/caesarivs/.config/sublime-text-3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.084944
startup time: 0.225292
first paint time: 0.241406
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_file_settings
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.quick_panel
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-ssl-linux
Linux SSL: successfully loaded _ssl module for libssl.so.1.0.0
reloading plugin Javatar.Javatar
reloading plugin Package Control.Package Control
reloading plugin Package Control.bootstrap
plugins loaded
Javatar snippet Class.javatar loaded
Javatar snippet Enumerator.javatar loaded
Javatar snippet Interface.javatar loaded
reloading Packages/User/Javatar.sublime-settings
[Javatar] Use default Java version [JDK1.7.0u79]
reloading Packages/User/Javatar.sublime-settings
Package Control: No updated packages
reloading Packages/User/Javatar.sublime-settings
reloading Packages/User/Javatar.sublime-settings
reloading Packages/User/Javatar.sublime-settings
reloading Packages/User/Javatar.sublime-settings
reloading Packages/User/Javatar.sublime-settings
reloading Packages/User/Javatar.sublime-settings
```
And javatar's user settings:

```
{
	"javatar_gp": 0,
	"jdk_version":
	{
		"use": ""
	},
	"message_id": 20,
}
```

**Comment by spywhere** [comment]
From the @m-delvalle comment, I think the problem cause by the folder name case (in the code it's `user` not `User`) which the OS might be case-sensitive system. The workaround until the fix is pushed is to download the JavaSE8 package and install it manually (to `User` directory *not* `Installed Packages`).

**Comment by spywhere** [comment]
The issue has been fixed in Javatar v1.0.2. If you still get the problem please reopen the issue again.
